axelor-base/src/main/java/com/axelor/apps/base/service/printing/template/PrintingTemplateServiceImpl.java

```diff
@@ -38,21 +48,88 @@ public PrintingTemplateServiceImpl(PrintingTemplateRepository printingTemplateRe
   }
 
   @Override
-  public List<PrintingTemplate> getPrintingTemplates(String modelName) throws AxelorException {
+  public List<PrintingTemplate> getActivePrintingTemplates(String modelName)
+      throws AxelorException {
 
-    List<PrintingTemplate> list =
-        printingTemplateRepository
-            .all()
-            .filter("self.metaModel.fullName = :modelName")
-            .bind("modelName", modelName)
-            .fetch();
+    List<PrintingTemplate> templates =
+        getPrintTemplates(
+            modelName, List.of(PrintingTemplateRepository.PRINTING_TEMPLATE_STATUS_SELECT_ACTIVE));
 
-    if (CollectionUtils.isEmpty(list)) {
+    if (CollectionUtils.isEmpty(templates)) {
       throw new AxelorException(
           TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
           I18n.get(BaseExceptionMessage.TEMPLATE_CONFIG_NOT_FOUND));
     }
 
-    return list;
+    return templates;
+  }
+
+  @Override
+  public void addPrintButton(String model, Response response) {
+    if (Boolean.FALSE.equals(hasActivePrintingTemplates(model))
+        || !isValidViewForToolBar(response.getData())) {
+      return;
+    }
+    try {
+      List<Button> toolbar = getToolbar(response);
+      if (toolbar.stream().map(Button::getName).noneMatch(PRINT_BTN_NAME::equals)) {
+        toolbar.add(getPrintBtn());
+      }
+    } catch (IllegalAccessException e) {
+      TraceBackService.trace(e);
+    }
+  }
+
+  @Override
+  public Boolean hasActivePrintingTemplates(String modelName) {
```

Comment:
I think you can return a `boolean` here, to match the type of ObjectUtils.notEmpty